src/common/include/sirf/common/GeometricalInfo.h

```diff
@@ -117,6 +118,34 @@ class VoxelisedGeometricalInfo :
 	Spacing _spacing;
 	Size _size;
 	DirectionMatrix _direction;
+	static bool near_(const Coordinate& x, const Coordinate& y, float eps)
+	{
+		float s = 0;
+		float t = 0;
+		for (int i = 0; i < num_dimensions; i++) {
+			float xi = x[i];
+			float yi = y[i];
+			s = std::max(s, std::abs(xi));
+			s = std::max(s, std::abs(yi));
+			t = std::max(t, std::abs(xi - yi));
+		}
+		return t <= eps * s;
```

Comment:
This near function just looks at the maximum distance between two `Coordinate` instances and not the element-wise comparison, right?

so if `x = [100.0, 0.1, 3.0]` and `y=[100.0, 1.0, 3.0]` then `s` should be `100` and `t = 0.9` so then if you call it with `eps= 1e-2` the condition reads true because `0.9 < 1e-2 * 100` but the two Coordinates are in fact different... or did I read the code wrong?

same goes for the comparison of direction matrices below

Reply:
Shouldn't there be an element-wise float comaprison between two floats a and b as `equal(a,b,eps)= abs(a-b) <= eps*max(a,b)`... then it would read for the above example for the 2nd component that `0.9 < 1e-2 * 1` and that would evaluate to `false` (which is not a proof of course :D )

Reply:
ok. good point! In fact, there is no need to doing a relative comparison here. We want to compare distances in up to .01 mm, and directions vectors up to 0.1.

We can therefore simply use `norm(x-y)<=eps`

Reply:
@johannesmayer but if `x[0] = 0` becomes `y[0] = 1e-7` purely because of the round-off error in calculating `y` then `near_` would return `false`.

@KrisThielemans I went for relative distances because I was not sure about measurement units, switched to absolute now.

Reply:
sure. looks fine now